### PR TITLE
mu: 1.0 -> 1.2

### PR DESCRIPTION
**Pull request by jabranham**
###### Motivation for this change
Upstream release

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [X] Assured whether relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by xeji** [comment]
@GrahamcOfBorg build mu

File: pkgs/tools/networking/mu/default.nix [hunk 1]
```diff
@@ -1,31 +1,26 @@
...
     repo   = "mu";
-    rev    = "v${version}";
-    sha256 = "0y6azhcmqdx46a9gi7mn8v8p0mhfx2anjm5rj7i69kbr6j8imlbc";
+    rev    = version;
+    sha256 = "0yhjlj0z23jw3cf2wfnl98y8q6gikvmhkb8vdm87bd7jw0bdnrfz";
   };
 
-  # 0.9.18 and 1.0 have 2 failing tests but previously we had no tests
-  patches = [
-    ./failing_tests.patch
```
Comment (peterhoeg) [inline pkgs/tools/networking/mu/default.nix]: If we no longer need this, please remove the file as well.

Comment (jabranham) [reply pkgs/tools/networking/mu/default.nix]: Indeed. Done, thanks!

**Comment by xeji** [comment]
@GrahamcOfBorg build mu

**Comment by xeji** [comment]
aarch64 timed out building rustc, no point in waiting for that....

**Review by xeji — APPROVED** [review]
LGTM and works locally on x86_64-linux

**Pull request merged**